app/components/uploadlaunchoverlay/uploadlaunchoverlay.js

```diff
@@ -0,0 +1,123 @@
+/**
+ * Copyright (c) 2017, Tidepool Project
+ *
+ * This program is free software; you can redistribute it and/or modify it under
+ * the terms of the associated License, which is identical to the BSD 2-Clause
+ * License as published by the Open Source Initiative at opensource.org.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the License for more details.
+ *
+ * You should have received a copy of the License along with this program; if
+ * not, you can obtain one from Tidepool Project at tidepool.org.
+ */
+
+import _ from 'lodash';
+import React, { Component } from 'react';
+import cx from 'classnames';
+import GitHub from 'github-api';
+import ModalOverlay from '../modaloverlay';
+import { URL_UPLOADER_DOWNLOAD_PAGE } from '../../core/constants';
+import logoSrc from '../uploaderbutton/images/T-logo-dark-512x512.png';
+
+const github = new GitHub();
+
+class UploadLaunchOverlay extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      latestWinRelease: null,
+      latestMacRelease: null,
+      error: null
+    };
+  }
+
+  static propTypes = {
+    overlayClickHandler: React.PropTypes.func.isRequired
+  };
+
+  componentWillMount = () => {
+    const uploaderRepo = github.getRepo('tidepool-org/chrome-uploader');
+    uploaderRepo.listReleases((err, releases, request) => {
+      if(err){
+        this.setState({error: true});
+      }
+      this.setState(this.getLatestGithubRelease(releases));
+    });
+  }
+
+  getLatestGithubRelease = (releases) => {
+    const latestRelease = _.filter(releases, {prerelease: false})[0];
```

Comment:
I had a similar thought, but realized that it wouldn't be able to generate any links at all until we had an Electron version released. Since we _should_ always wait until all the relevant artifacts are attached to a GitHub release before attaching the tag and publishing it, I feel like it's a relatively low risk to presume that the URL exists. If human error creates a situation in which this becomes an issue in the future (and we have releases which have these artifacts at all), then I think we could add some additional checks.